### PR TITLE
teku 21.12.2 (new formula)

### PR DESCRIPTION
**Pull request by riptl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Teku is an implementation of the Eth2 beacon chain in Java. Eth2 is a series of ongoing upgrades to the Ethereum cryptocurrency to fix the high energy usage of the network.

This PR is part of a series of new formulas. There exist a few implementations of Eth2, most notably Prysm (Go), Nimbus-Eth2 (Nim) #91969, Lighthouse (Rust) #91984 and Teku (Java). All of these clients are well maintained, implement the same protocol and have mostly the same ergonomics. I aim to enroll all of them in Homebrew.

### CODE REVIEW

File: Formula/teku.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    EOS
+
+    (buildpath/"teku.args").write <<~EOS
+      --data-path=#{var}/teku
+      --log-destination=FILE
+      --log-file=#{var}/log/teku.log
+      --rest-api-enabled
+    EOS
+
+    etc.install "teku.args"
```
Comment (SMillerDev) [inline Formula/teku.rb:30]: We do not want to run our own configuration here. We only ship what upstream ships

Comment (terorie) [reply Formula/teku.rb:30]: I can remove `--rest-api-enabled`, but the rest is specific to how the application is run with `brew services`. Is it okay to leave `data-path` and `log-*` in?

Comment (SMillerDev) [reply Formula/teku.rb]: No, all of this needs to be removed. If it can't run in the default configuration it can't have `brew services support`.

Comment (terorie) [reply Formula/teku.rb]: I see. I was using [prometheus](https://github.com/Homebrew/homebrew-core/blob/bae1ec534356c4afa485e797ab1883c48bdf8bd2/Formula/prometheus.rb#L42) as an example but I wasn't aware this was an exception.

Comment (terorie) [reply Formula/teku.rb]: I've removed all defaults and replaced it with an empty YAML config file.

Comment (SMillerDev) [reply Formula/teku.rb]: Yeah, I'm not sure how Prometheus was ever accepted like that

Comment (terorie) [reply Formula/teku.rb]: Hm, I've removed the `brew services` section for now. The empty-YAML hack doesn't work either, so this software is not adequate to be run in compliance with the services rules.

**Review by chenrui333 — APPROVED** [review]

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Thanks @terorie !

**Pull request closed without merging**